# Airbnb New User Bookings Analysis

## 1. Background

Airbnb was founded in 2008 and has become a giant in the short-term rental homestay industry in just 12 years, and it has strongly impacted on the traditional hotel industry.

Airbnb has presented in 191 countries and often appears in many business analysis case studies. It is obvious that Airbnb has done an amazing job on product experience, marketing strategy, and customer service, it always interested to know if there is still any room for improvement or how to bring more value to the business. 

## 2. Purpose

The purpose of the following analysis can be broken down into three major directions and four business questions:

### Three Major Analysis Directions：

* User Profile Analysis

* Affiliate Channels Analysis

* Funnel Model Analysis

###  Business Question：

* What are the users' characteristics of registered Airbnb users?

* How has Airbnb’s users growth trend and conversion rate changed?

* What are the deficiencies in Airbnb’s affiliate channel ? Which promotion channels bring better conversion rate?

* How is the conversion rate in every users' activity ? Where need to be improved?

## 3. Data Description

* Data Source : https://www.kaggle.com/c/airbnb-recruiting-new-user-bookings/data

* Data was extracted from 2010 to 2014 which recorded Airbnb users' activties 

The dataset that was used to do the following analysis contains two tables: TRAIN_URSERS and SESSION. Below is the detail description of the column features. 

<img src="table.PNG">

## 4. Data Cleaning

### Check missing value 

In [4]:
import pymysql
%load_ext sql
%sql mysql+pymysql://root:ansonandtan1213@localhost/airbnb

In [37]:
# Check missing value in train_users 
%sql SELECT SUM(CASE WHEN date_first_booking IS NULL THEN 1 ELSE 0 END) date_first_booking,
            SUM(CASE WHEN gender = '-unknown-' THEN 1 ELSE 0 END) gender,
            SUM(CASE WHEN age IS NULL THEN 1 ELSE 0 END) age,
            SUM(CASE WHEN first_affiliate_tracked IS NULL THEN 1 ELSE 0 END) first_affiliate_tracked,
            SUM(CASE WHEN first_browser ='-unknown-' THEN 1 ELSE 0 END) first_browser 
            FROM train_users

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(Decimal('124543'), Decimal('95688'), Decimal('87990'), Decimal('6065'), Decimal('27266'))]

In [18]:
# Check missing value in session  
%sql select SUM(CASE WHEN action_type IS NULL THEN 1 ELSE 0 END) action_type,
            SUM(CASE WHEN action_detail IS NULL THEN 1 ELSE 0 END) action_detail 
            FROM session 

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(Decimal('1126204'), Decimal('1126204'))]

#### Reason for missing value and how to handle. 

* Missing value in date_first_booking can be explained that a user hasn't made any booking. 
* Gender & Age is optional, missing value means user did not provide those information. 
* The other missing values can be caused by data collection stage. 

Method: Filter out the null value in the following analysis

### Check duplicate data

In [23]:
%sql SELECT id, COUNT(id) AS count_id FROM train_users GROUP BY id HAVING count_id > 1;

 * mysql+pymysql://root:***@localhost/airbnb
0 rows affected.


[]

There is no duplicate data in train_users

### Handling the outliers

In [41]:
%sql select min(age),max(age) from train_users

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(1, 2014)]

Only select the row where users' age in between 10 - 80 

In [44]:
%sql CREATE TABLE users AS (SELECT * FROM train_users WHERE age <= 80 and age >=10) # created new table to store the result. 

 * mysql+pymysql://root:***@localhost/airbnb
122633 rows affected.


[]

## 5. User Behavior Analysis


###  5.1 User profile analysis


####  5.1.1 Gender Distribution 

In [7]:
%sql SELECT gender,COUNT(*),
CONCAT(ROUND((COUNT(*)/(SELECT count(*) FROM users WHERE gender ='male' or gender = 'female')*100),1),'%') AS  'Percentage' 
FROM  users WHERE gender ='male' or gender = 'female' GROUP BY gender

 * mysql+pymysql://root:***@localhost/airbnb
2 rows affected.


[('MALE', 49785, '46.8%'), ('FEMALE', 56536, '53.2%')]

<img src="gender.PNG">

Female registered users are 7.32% more than Male registered users.

####  5.1.2 User's Age Distribution 

In [72]:
%sql SELECT age_group,number_users,CONCAT(ROUND((number_users/(SELECT COUNT(*) FROM users))*100,2),'%') AS percentage 
FROM
(
    SELECT CASE WHEN age BETWEEN 10 AND 20 THEN '10-20' 
                WHEN age BETWEEN 21 AND 30 THEN '21-30' 
                WHEN age BETWEEN 31 AND 40 THEN '31-40' 
                WHEN age BETWEEN 41 AND 50 THEN '41-50' 
                WHEN age BETWEEN 51 AND 60 THEN '51-60' 
                WHEN age BETWEEN 61 AND 70 THEN '61-70' 
                WHEN age BETWEEN 71 AND 80 THEN '71-80' 
                ELSE NULL 
                END AS age_group,COUNT(*) AS number_users 
                FROM users 
                GROUP BY age_group 
                ORDER BY age_group) t

 * mysql+pymysql://root:***@localhost/airbnb
7 rows affected.


[('10-20', 2412, '1.97%'),
 ('21-30', 41633, '33.95%'),
 ('31-40', 44212, '36.05%'),
 ('41-50', 18831, '15.36%'),
 ('51-60', 9868, '8.05%'),
 ('61-70', 4656, '3.80%'),
 ('71-80', 1021, '0.83%')]

<img src="age.PNG">

The above age distribution indicated that the main users of Airbnb are in Young and Middle-aged group. Among them, the 80s(29-39 years old) have the largest number of users, followed by 90s and 70s.



#### 5.1.3 Users' Devices Distribution 

In [73]:
%sql SELECT first_device_type, COUNT(*) AS number_users, 
    CONCAT(ROUND(count(*)/(select count(*) from users)*100,1),'%') as percentage 
    FROM users 
    GROUP BY first_device_type;

 * mysql+pymysql://root:***@localhost/airbnb
9 rows affected.


[('Mac Desktop', 56973, '46.5%'),
 ('Windows Desktop', 41128, '33.5%'),
 ('iPhone', 10425, '8.5%'),
 ('Other/Unknown', 3937, '3.2%'),
 ('iPad', 7515, '6.1%'),
 ('Desktop (Other)', 738, '0.6%'),
 ('Android Phone', 1244, '1.0%'),
 ('Android Tablet', 634, '0.5%'),
 ('SmartPhone (Other)', 39, '0.0%')]

<img src="device.PNG">

Since the data was obtained between 2010 and 2014, Mobile app was not yet became the pirmary device for the users.The propotion of Desktop registered users are much higher than Mobile users，Mac and Windows take about 76%. Specifically, Iphone Devices taken much higher portion than Android Devices. 

#### 5.1.4 Destination Coutry Distribution 

In [84]:
%sql SELECT country_destination, COUNT(*) AS number_users, CONCAT(ROUND(count(*)/(SELECT count(*) FROM users WHERE country_destination <> 'ndf' )*100,1),'%') AS percentage 
FROM users 
WHERE country_destination <> 'ndf'
GROUP BY country_destination 
ORDER BY percentage desc ;

 * mysql+pymysql://root:***@localhost/airbnb
11 rows affected.


[('US', 47589, '71.0%'),
 ('FR', 3646, '5.4%'),
 ('IT', 1977, '2.9%'),
 ('GB', 1731, '2.6%'),
 ('ES', 1673, '2.5%'),
 ('other', 7391, '11.0%'),
 ('CA', 1054, '1.6%'),
 ('DE', 833, '1.2%'),
 ('NL', 589, '0.9%'),
 ('AU', 428, '0.6%'),
 ('PT', 156, '0.2%')]

<img src="destination.PNG">

Most users choose domestic travel which accounting for 70.16%. In terms of travel abroad, France has the highest proportion, followed by Italy, and other countries have a relatively small proportion

### 5.2 Users' Metric Analysis



#### 5.2.1 User growth trend and  conversion rate


In [8]:
%sql SELECT DATE_FORMAT(date_account_created,'%y-%m') AS Year_Mo,
COUNT(*) AS new_users, SUM(CASE WHEN date_first_booking IS NOT NULL THEN 1 ELSE 0 END) AS paid_users,
CONCAT(ROUND((SUM(CASE WHEN date_first_booking IS NOT NULL THEN 1 ELSE 0 END)/COUNT(*))*100,2),'%') AS Convertion 
FROM train_users 
GROUP BY date_format(date_account_created,'%y-%m') 
ORDER BY date_format(date_account_created,'%y-%m') 
LIMIT 10 

 * mysql+pymysql://root:***@localhost/airbnb
10 rows affected.


[('10-01', 42, Decimal('24'), '57.14%'),
 ('10-02', 63, Decimal('36'), '57.14%'),
 ('10-03', 82, Decimal('58'), '70.73%'),
 ('10-04', 87, Decimal('52'), '59.77%'),
 ('10-05', 125, Decimal('79'), '63.20%'),
 ('10-06', 116, Decimal('66'), '56.90%'),
 ('10-07', 183, Decimal('122'), '66.67%'),
 ('10-08', 169, Decimal('115'), '68.05%'),
 ('10-09', 217, Decimal('144'), '66.36%'),
 ('10-10', 204, Decimal('133'), '65.20%')]

<img src="montlyuser.PNG">

From the bar chart above, we can see airbnb had maintain a steady new users growth rate. The peak of the trend is concentrated in the third quarter of each year, from July to October, and there is a seasonal pattern of growth. It is speculated that people prefer to travel in summer, driving the growth of airbnb business.


#### 5.2.2 Average time it took for a new user to make the first reservation



In [12]:
%sql SELECT TIMESTAMPDIFF(MONTH, date_account_created, date_first_booking) AS Month_diff,
count(*) AS number_users, 
CONCAT(ROUND(count(*)/(select count(*) from users where date_first_booking is not null)*100,1),'%') as Percentage 
FROM users
WHERE date_first_booking is not null
GROUP BY Month_diff 
HAVING Month_diff >= 0
ORDER BY Month_diff

 * mysql+pymysql://root:***@localhost/airbnb
13 rows affected.


[(0, 42552, '63.4%'),
 (1, 8319, '12.4%'),
 (2, 2710, '4.0%'),
 (3, 1887, '2.8%'),
 (4, 1728, '2.6%'),
 (5, 1407, '2.1%'),
 (6, 1435, '2.1%'),
 (7, 1339, '2.0%'),
 (8, 1286, '1.9%'),
 (9, 1296, '1.9%'),
 (10, 1348, '2.0%'),
 (11, 1098, '1.6%'),
 (12, 643, '1.0%')]

Over 80% of users made their first booking within 3 months after they signed up for the account. 

## 6. Analysis of Affiliate Channels


#### 6.1 Number of registered users and  conversion rate of different affiliate channels


In [16]:
%sql SELECT CONCAT(affiliate_channel, '_', affiliate_provider) AS Affiliate, 
COUNT(id) AS new_user,SUM(case when date_first_booking IS NOT NULL THEN 1 ELSE 0 END) AS paid_user,
CONCAT(ROUND(SUM(CASE WHEN date_first_booking IS NOT NULL THEN 1 ELSE 0 END) / COUNT(id)*100,1),'%') AS convertion  
FROM users 
GROUP BY Affiliate 
Having new_user > 200 
ORDER BY new_user DESC;

 * mysql+pymysql://root:***@localhost/airbnb
19 rows affected.


[('direct_direct', 78645, Decimal('44922'), '57.1%'),
 ('sem-brand_google', 14656, Decimal('8081'), '55.1%'),
 ('sem-non-brand_google', 8916, Decimal('4583'), '51.4%'),
 ('api_other', 5441, Decimal('2292'), '42.1%'),
 ('seo_google', 3384, Decimal('2086'), '61.6%'),
 ('other_other', 2249, Decimal('1153'), '51.3%'),
 ('other_craigslist', 2196, Decimal('1145'), '52.1%'),
 ('content_google', 1433, Decimal('309'), '21.6%'),
 ('seo_facebook', 1167, Decimal('587'), '50.3%'),
 ('remarketing_google', 591, Decimal('277'), '46.9%'),
 ('sem-brand_bing', 495, Decimal('252'), '50.9%'),
 ('other_padmapper', 454, Decimal('208'), '45.8%'),
 ('content_facebook', 421, Decimal('60'), '14.3%'),
 ('sem-non-brand_bing', 421, Decimal('195'), '46.3%'),
 ('other_facebook-open-graph', 400, Decimal('123'), '30.8%'),
 ('sem-non-brand_vast', 380, Decimal('181'), '47.6%'),
 ('seo_bing', 256, Decimal('123'), '48.0%'),
 ('seo_yahoo', 246, Decimal('112'), '45.5%'),
 ('seo_other', 204, Decimal('118'), '57.8%')]

<img src="AFFI.PNG">

Direct (Direct website visit) has the largest number of registered users, with 137,403 users, accounting for 64.38%.

Since direct access from the official website is more convenient than other methods and over 80% users used desktop as their primary device, the number of users from "direct-direct" is much higher than other channels, and the conversion rate is as high as 57.1%, but since the direct website visit channel does not involve with marketing, it should be excluded when doing the analysis.



* Note：Affiliate Channels with less than 200 registrations are excluded from the chart below


<img src="affi2.PNG">

It can be seen from the charts above:


* After excluding direct, sem-brand-google (Search Engine Marketing by Google) accounted for 33.84% of new users. Overall, it complies with the Pareto principle. The top five channels contribute nearly 80% of registered users.


* Airbnb maintained the overall conversion rate very well where most of the channels' conversion rate is above 45%


* Under the top 5 affliate channels,the API channel with the conversion rate of 42.1% which is the lowest among the top 5. 


* The conversion rate of Content type channel performed worse than the others. Especially, content_google and content_facebook are only accounted for 21.6% and 14.3% respectively, which are far lower than the average conversion rate.


* In view of the performance of the above channels, based on the Pareto Principle, resources can be more focused on the main marketing channels of'sem-brand-google','sem-non-brand-google', and'seo-google'. 


#### 6.2  Number of registered users and  conversion rate of different marketing that users interacted with before signing up


In [19]:
%sql SELECT first_affiliate_tracked as first_marketing, 
COUNT(id) AS new_user,SUM(case when date_first_booking IS NOT NULL THEN 1 ELSE 0 END) as paid_user,
CONCAT(ROUND(SUM(CASE WHEN date_first_booking IS NOT NULL THEN 1 ELSE 0 END) / COUNT(id)*100,1),'%') AS convertion 
FROM users 
WHERE first_affiliate_tracked is not null 
GROUP BY first_affiliate_tracked

 * mysql+pymysql://root:***@localhost/airbnb
7 rows affected.


[('untracked', 63753, Decimal('36743'), '57.6%'),
 ('linked', 27810, Decimal('15478'), '55.7%'),
 ('omg', 24444, Decimal('12239'), '50.1%'),
 ('tracked-other', 3750, Decimal('1807'), '48.2%'),
 ('product', 797, Decimal('380'), '47.7%'),
 ('marketing', 101, Decimal('61'), '60.4%'),
 ('local ops', 26, Decimal('7'), '26.9%')]

<img src="marketing.PNG">

The 'untracked' method should be excluded when performing the analysis even though it has the high conversion rate and numbers of new users. Both the 'linked' and'omg' marketing strategies account for a relatively high user conversion rate, which can be highly promoted while 'local ops' ranks the last of all marketing methods, the amount of traffic and conversion are not ideal, the allocation of the related resources can be reduced.


## 7. Funnel Model Analysis


In [23]:
# Total Users
%sql SELECT COUNT(*)  AS 'Total Users' FROM (SELECT user_id FROM session GROUP BY user_id) new_sessions;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(135479,)]

In [28]:
# Activate Users：Based on to the total number of activities of the user, if the user has 10 times or more activities, it can indicate that the user is an active user.

%sql SELECT COUNT(*)  AS 'Total Activate User' 
FROM (SELECT user_id FROM session GROUP BY user_id HAVING COUNT(user_id) >= 10) active;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(112016,)]

In [5]:
# Registered Users
%sql SELECT COUNT(*) AS 'Total Registered Users'  
FROM (SELECT user_id FROM session GROUP BY user_id) new_sessions  INNER JOIN train_users tu  ON new_sessions.user_id = tu.id;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(73812,)]

In [34]:
# Booked Users
%sql SELECT COUNT(*) AS 'Total Booked Users' 
FROM ( SELECT user_id FROM session WHERE action_detail = 'reservations' GROUP BY user_id) booking;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(10367,)]

In [36]:
# Actual Paid Users
%sql SELECT COUNT(*) AS 'Total Actual Paying Users' 
FROM ( SELECT user_id FROM session WHERE action_detail = 'payment_instruments' GROUP BY user_id) payed;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(9019,)]

In [37]:
# Repurchased Users
%sql SELECT COUNT(*) AS 'Total repurchase users' 
FROM (SELECT user_id FROM session WHERE action_detail = 'reservations' GROUP BY user_id HAVING COUNT(user_id) >= 2) re_booking;

 * mysql+pymysql://root:***@localhost/airbnb
1 rows affected.


[(5447,)]

<img src="funnel.PNG">

It can be seen from the funnel chart that the number of repurchased users only accounted for 4.0% of the total number of users. Customers are most likely to churn when made their bookings after registered.There are 73,812 registered users, but the actual users who place orders only account for about 7.7%. The optimazation need to be done through more careful data research and perform in-depth analysis.  

##  8. Summary and Recommendation 



### 8.1 Summary




* Among the registered users, male users accounted for 46.8% while female users accounted for 53.2%, the difference between the two are not significant. 


* Users' age are wildely distributed, However, over 80% of users'age are between 26 and 39 years old. The 80s users are the main user group, accounting for nearly 40%


* The proportion of desktop devices usage is much higher than mobile devices, with Mac and Windows devices accounting for about 80%. The proportion of Apple devices is much higher than Android devices


* The user growth trend is flat in the early stage, rapid growth in the middle and late stages, overall,it maintained a very health upward trend. The peak happened in the third quarter of each year, from July to October, and there is a seasonal pattern of the growth. 


* The early stage conversion rate was maintained over 60%, with a peak at 83% on April, 2011 which showing the excellent performance. With the increase in the number of registered users on the platform, the conversion rate had gradually decreased and remained at the level of 50% after the initial bonus period. 


*  The top five affiliate channels account for nearly 80% of registered users. The conversion rate is relatively ideal, within the top five channels, only API channel is lower than the average conversion rate within the top five channels. 


* Google affiliate channels bring much more users than other channels. 


* The churn rate is largest  with a conversion rate of only 14.04%.


* The largest customer loss happened when customer registered until they actually made the booking. 

### 8.2 Suggestion 

* Based on user profile, the target audience can be aimed at women born in the 1980s, while considering moving closer to younger generations.


* There is a growth peak in every summer. Marketing related operation can mainly focus on s during the summer peak travel season, while strengthening promotion in autumn and winter in order to increase user retention.


* Content affiliate channels can be improved through better content operation and management.


* Most of the user registered through Google channels, and we can continue to strengthen the investment on Google marketing. 


* Reducing customer churn and keep them loyal by offering promotion or incentive for making new bookings, and increase the number of repurchase users by building relationships with the existing customers. 